src/System/Wlog/Concurrent.hs

```diff
@@ -14,31 +14,32 @@ module System.Wlog.Concurrent
 
 import Universum
 
-import Control.Concurrent (threadDelay)
 import Control.Concurrent.Async.Lifted (withAsyncWithUnmask)
 import Control.Monad.Trans.Control (MonadBaseControl)
-import Data.Time.Units (Microsecond, Second, convertUnit, toMicroseconds)
-import Fmt
+import Fmt ((+|), (+||), (|+), (||+))
+import GHC.Real ((%))
+import Time (Microsecond, RatioNat, Second, Time, threadDelay, timeMul, toUnit)
 
 import System.Wlog.CanLog (WithLoggerIO, logWarning)
 
 -- | Data type to represent waiting strategy for printing warnings
 -- if action take too much time.
 data WaitingDelta
-    = WaitOnce      Second              -- ^ wait s seconds and stop execution
-    | WaitLinear    Second              -- ^ wait s, s * 2, s * 3  , s * 4  , ...      seconds
-    | WaitGeometric Microsecond Double  -- ^ wait m, m * q, m * q^2, m * q^3, ... microseconds
+      -- | wait s seconds and stop execution
+    = WaitOnce      (Time Second)
+      -- | wait s, s * 2, s * 3  , s * 4  , ...      seconds
+    | WaitLinear    (Time Second)
+      -- | wait m, m * q, m * q^2, m * q^3, ... microseconds
+    | WaitGeometric (Time Microsecond) RatioNat
```

Comment:
You can store `Time Second` here as well. `Microsecond` was used here because multiplying integer seconds and some `Double` won't work.